docs/README.md

```diff
@@ -0,0 +1,69 @@
+# Koog Documentation
+
+This module contains documentation for the Koog framework, including user guides, API references, prompting guidelines, and other static files.
+
+## Module Structure
+
+The docs module is organized as follows:
+
+- **docs/** - Contains markdown files with user documentation
+- **overrides/** - Custom overrides for the MkDocs theme
+- **prompt/** - Prompting guidelines with extensions for popular modules
+- **src/** - Knit generated source code from documentation code snippets, should not be commited
+
+## Documentation System
+
+### MkDocs
+
+The documentation is built using [MkDocs](https://www.mkdocs.org/) with the Material theme. The configuration is defined in `mkdocs.yml`, which specifies:
+
+- Navigation structure
+- Theme configuration
+- Markdown extensions
+- Repository links
+
+The documentation is available at [https://docs.koog.ai/](https://docs.koog.ai/).
+
+### Docs Code Snippets Verification
+
+To ensure code snippets in documentation are compilable and up-to-date with the latest framework version, the [kotlinx-knit](https://github.com/Kotlin/kotlinx-knit) library is used.
+
+Knit provides a Gradle plugin that extracts specially annotated Kotlin code snippets from markdown files and generates Kotlin source files. 
+To extract code snippets, run:
+```
+./gradlew :docs:knit
+```
+
+To verify that all extracted files are compilable, run:
+```
+./gradlew :docs:build
+```
+
+**NB**: Before every release follow these actions:
```

Comment:
is it possible to always build the docs against the sources? then, at the moment of a release, docs will be already ready

Reply:
hm, maybe, then we should use there koog from repo and every time we update the api change the docs. Sounds reasonable

Reply:
yep, docs will be always in sync with the sources

Reply:
Added knit as a check. It also allowed to remove `koog` dependency and use project dependency from repo! win-win